pkg/generators/list.go

```diff
@@ -16,10 +18,34 @@ func NewListGenerator() Generator {
 	return g
 }
 
-func (g *ListGenerator) GenerateApplications(appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
-	if appSet == nil {
+func (g *ListGenerator) GenerateApplications(appSetGenerator *argoprojiov1alpha1.ApplicationSetGenerator,
+	appSet *argoprojiov1alpha1.ApplicationSet) ([]argov1alpha1.Application, error) {
+	if appSetGenerator == nil || appSet == nil {
 		return nil, fmt.Errorf("ApplicationSet is empty")
 	}
 
+	if appSetGenerator.List == nil {
+		return nil, nil
+	}
+
+	listGenerator := appSetGenerator.List
+
+	if listGenerator == nil {
+		return nil, fmt.Errorf("There isn't list generator ")
+	}
+
+	var tmplApplication argov1alpha1.Application
+	tmplApplication.Namespace = appSet.Spec.Template.Namespace
+	tmplApplication.Name = appSet.Spec.Template.Name
+	tmplApplication.Spec = appSet.Spec.Template.Spec
+
+	params := make(map[string]string, 2)
+	for _, tmpItem := range listGenerator.Elements {
+		params[utils.ClusterListGeneratorKeyName] = tmpItem.Cluster
+		params[utils.UrlGeneratorKeyName] = tmpItem.Url
+		tmpApplication, err := utils.RenderTemplateParams(&tmplApplication, params)
+		log.Infof("tmpApplication %++v", tmpApplication)
+		log.Infof("error %v", err)
```

Comment:
Suggest something like if err != nil { log.WithError(err).Error("error rendering template params") }

Reply:
This is only for debug, I will check error later